app/controllers/OtherReasonController.scala

```diff
@@ -226,8 +265,13 @@ class OtherReasonController @Inject()(whenDidBecomeUnablePage: WhenDidBecomeUnab
                     logger.debug("[OtherReasonController][removeFileUpload] - No files left in journey - redirecting to first document upload page")
                     Redirect(controllers.routes.OtherReasonController.onPageLoadForFirstFileUpload(mode))
                   } else {
-                    //TODO: placeholder until 'You have uploaded X files' has been implemented (PRM-796)
-                    Ok("you have uploaded X files")
+                      val formProvider: Form[String] = FormProviderHelper.getSessionKeyAndAttemptToFillAnswerAsString(
+                        YouHaveUploadedFilesForm.youHaveUploadedForm,
+                        SessionKeys.nextFileUpload
+                      )
+                      val radioOptionsToRender: Seq[RadioItem] = RadioOptionHelper.yesNoRadioOptions(formProvider)
+                      val postAction = controllers.routes.OtherReasonController.onSubmitForUploadComplete()
+                        Ok(youHaveUploadedFilesPage(formProvider, radioOptionsToRender, postAction, amountOfFiles))
```

Comment:
You don't actually need to do this, there is a much faster way.
We can do 
```suggestion
                      Redirect(controllers.routes.OtherReasonController.onPageLoadForUploadComplete(mode))
```

Reply:
Done.